music_assistant/server/providers/deezer/helpers.py

```diff
@@ -0,0 +1,576 @@
+    """Async wrapper of the deezer-python add_user_artists function."""
+
+    def _get_artist():
+        success = client.add_user_artist(artist_id=artist_id)
+        return success
+
+    return await asyncio.to_thread(_get_artist)
+
+
+async def remove_user_artists(client: deezer.Client, artist_id: int) -> bool:
+    """Async wrapper of the deezer-python remove_user_artists function."""
+
+    def _get_artist():
+        success = client.remove_user_artist(artist_id=artist_id)
+        return success
+
+    return await asyncio.to_thread(_get_artist)
+
+
+async def search_album(client: deezer.Client, query: str, limit: int = 5) -> list[deezer.Album]:
+    """Async wrapper of the deezer-python search_albums function."""
+
+    def _search():
+        result = client.search_albums(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def search_track(client: deezer.Client, query: str, limit: int = 5) -> list[deezer.Track]:
+    """Async wrapper of the deezer-python search function."""
+
+    def _search():
+        result = client.search(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def search_artist(client: deezer.Client, query: str, limit: int = 5) -> list[deezer.Artist]:
+    """Async wrapper of the deezer-python search_artist function."""
+
+    def _search():
+        result = client.search_artists(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def search_playlist(
+    client: deezer.Client, query: str, limit: int = 5
+) -> list[deezer.Playlist]:
+    """Async wrapper of the deezer-python search_playlist function."""
+
+    def _search():
+        result = client.search_playlists(query=query)[:limit]
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def get_access_token(mass, app_id, app_secret, code) -> Credential:
+    """Update the access_token."""
+    response = await _post_http(
+        mass=mass,
+        url="https://connect.deezer.com/oauth/access_token.php",
+        data={
+            "code": code,
+            "app_id": app_id,
+            "secret": app_secret,
+        },
+        params={
+            "code": code,
+            "app_id": app_id,
+            "secret": app_secret,
+        },
+        headers=None,
+    )
+    try:
+        return response.split("=")[1].split("&")[0]
+    except Exception as error:
+        raise LoginFailed("Invalid auth code") from error
+
+
+async def _post_http(mass, url, data, params=None, headers=None) -> str:
+    async with mass.http_session.post(
+        url, headers=headers, params=params, json=data, ssl=False
+    ) as response:
+        if response.status != 200:
+            raise ConnectionError(f"HTTP Error {response.status}: {response.reason}")
+        response_text = await response.text()
+        return response_text
+
+
+def parse_artist(mass, artist: deezer.Artist) -> Artist:
```

Comment:
instead of passing in the whole mass instance you could just pass in the provider instance only but frankly I think this should reside in the provider class itself instead of passing it in.

If you want to keep it here, add annotations. You can use "TYPE_CHECKING" to import for type checking only and provide circular imports

Reply:
Yeah we moved it to the provider class